src/pages/client/auth/signUp/SignUp.tsx

```diff
@@ -1,26 +1,167 @@
-import React, { useRef } from 'react';
+import { Formik as FormValidation } from 'formik';
+import React from 'react';
 import ReCAPTCHA from 'react-google-recaptcha';
+import { CircularProgress } from '@mui/material';
 
+import './SignUp.scss';
+import Auth, { UserRoles } from '../../../../common/interfaces/Auth';
+import { AuthForm, ImageSide } from '../../../../components/AuthForm/AuthForm';
 import env from '../../../../common/config/interface/env';
-import { useLocalStorage } from '../../../../common/hooks/LocalStorage';
+import { CustomTextField, MuiButton, SubmitButtonStyle } from '../../../../components/MuiStyling/MuiStyling';
+import clientService from '../../../../services/clientService';
+import { CustomSnackbar } from '../../../../components/Snackbar/CustomSnackbar';
+interface SignUpFormInitValue {
+  firstName: string;
+  lastName: string;
+  email: string;
+  password: string;
+  reTypePassword?: string;
+  gCaptcha: string;
+  roles: UserRoles;
+}
 
 export const SignUp = () => {
-  const captchaRef = useRef<any>();
-  const [localValue, setLocalValue] = useLocalStorage('test', '');
-  const onSubmitWithReCAPTCHA = async (e: any) => {
-    e.preventDefault();
-    const token = await captchaRef.current?.executeAsync();
-    console.log('token: ', token);
-    // apply to form data
+  const captchaRef = React.useRef<any>();
+  const [loading, setLoading] = React.useState(false);
+  const [showSnackbar, setShowSnackbar] = React.useState(false);
+  const [responseFromAPI, setResponseFromAPI] = React.useState('');
+
+  const signUpSubmit = async (signUpValues: SignUpFormInitValue) => {
+    const gCaptcha: string = await captchaRef.current?.executeAsync();
+    const signUpObj = { ...signUpValues, gCaptcha };
+    delete signUpObj?.reTypePassword;
+
+    try {
+      setLoading(true);
+      const response = await clientService.userSignUp(signUpObj);
+      console.log('response: ', response);
+      setResponseFromAPI(response);
+      setShowSnackbar(true);
+    } catch (error: any) {
+      console.log('error: ', error.response?.data.message);
+      setResponseFromAPI(error.response?.data?.message);
```

Comment:
Refer error.response to a type

Reply:
I don't understand this comment. Can you explain clear for me?